Panic when `upload-dirs` is false

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Output of `ddev debug test`

```
$ ddev version
 ITEM             VALUE                                               
 DDEV version     v1.22.0-beta3-10-g42969991                          
 architecture     amd64                                               
 db               ddev/ddev-dbserver-mariadb-10.4:v1.22.0-beta1       
 ddev-ssh-agent   ddev/ddev-ssh-agent:v1.22.0-beta1                   
 docker           24.0.2                                              
 docker-compose   v2.20.0                                             
 docker-platform  dev                                                 
 mutagen          0.17.2                                              
 os               linux                                               
 router           traefik:v2.10                                       
 web              ddev/ddev-webserver:20230707_fix_process_group_kill
```

### Expected Behavior

From  `.ddev/config.yaml`, `upload_dirs` can be `false`:

```
# upload_dirs: "custom/upload/dir"
#
# upload_dirs:
#   - custom/upload/dir
#   - ../private
#
# upload_dirs: false
```

https://github.com/ddev/ddev/blob/42969991212ff2c71fb91d7651d2ea17f8d1d749/pkg/ddevapp/upload_dirs.go#L211-L212

### Actual Behavior

```
$ ddev config --auto --upload-dirs=false
Creating a new ddev project config in the current directory (/home/user/test) 
Once completed, your configuration will be written to /home/user/test/.ddev/config.yaml
 
Configuring unrecognized codebase as project of type 'php' at /home/user/test 
Configuration complete. You may now run 'ddev start'. 
panic: interface conversion: interface {} is bool, not ddevapp.UploadDirs

goroutine 1 [running]:
github.com/ddev/ddev/pkg/ddevapp.(*DdevApp).validateUploadDirs(0xc000103c00)
        /home/runner/work/ddev/ddev/pkg/ddevapp/upload_dirs.go:230 +0x485
github.com/ddev/ddev/pkg/ddevapp.(*DdevApp).LoadConfigYamlFile(0xc000103c00, {0xc000418db0?, 0xc00017b040?})
        /home/runner/work/ddev/ddev/pkg/ddevapp/config.go:363 +0x1b2
github.com/ddev/ddev/pkg/ddevapp.(*DdevApp).ReadConfig(0xc000103c00, 0x0)
        /home/runner/work/ddev/ddev/pkg/ddevapp/config.go:317 +0x4b
github.com/ddev/ddev/pkg/ddevapp.NewApp({0x0, 0x0}, 0x8?)
        /home/runner/work/ddev/ddev/pkg/ddevapp/config.go:112 +0x825
github.com/ddev/ddev/cmd/ddev/cmd.glob..func48(0x10f7500, {0xc000312340?, 0x0?, 0x2?})
        /home/runner/work/ddev/ddev/cmd/ddev/cmd/root.go:101 +0x57
github.com/spf13/cobra.(*Command).execute(0x10f7500, {0xc000312320, 0x2, 0x2})
        /home/runner/work/ddev/ddev/vendor/github.com/spf13/cobra/command.go:876 +0x770
github.com/spf13/cobra.(*Command).ExecuteC(0x10f5480)
        /home/runner/work/ddev/ddev/vendor/github.com/spf13/cobra/command.go:974 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
        /home/runner/work/ddev/ddev/vendor/github.com/spf13/cobra/command.go:902
github.com/ddev/ddev/cmd/ddev/cmd.Execute()
        /home/runner/work/ddev/ddev/cmd/ddev/cmd/root.go:163 +0x25
main.main()
        /home/runner/work/ddev/ddev/cmd/ddev/main.go:28 +0xc5
```

### Steps To Reproduce

1. `ddev config --auto --upload-dirs=false`

### Anything else?

_No response_

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -135,7 +135,7 @@ jobs:
       - name: Override environment variables for push-pull-test-platforms
         run: |
           echo "MAKE_TARGET=testpkg" >> $GITHUB_ENV
-          echo "TESTARGS=-failfast -run '(TestDdevFullSite.*|Test.*(Push|Pull))'" >> $GITHUB_ENV
+          echo "TESTARGS=-failfast -run '(TestDdevFullSite.*|TestDdevImportFiles|TestDdevAllDatabases|Test.*(Push|Pull))'" >> $GITHUB_ENV
           echo "GOTEST_SHORT=" >> $GITHUB_ENV
           echo "DDEV_PLATFORM_API_TOKEN=${{ secrets.DDEV_PLATFORM_API_TOKEN }}" >> $GITHUB_ENV
           echo "DDEV_PANTHEON_API_TOKEN=${{ secrets.DDEV_PANTHEON_API_TOKEN }}" >> $GITHUB_ENV

--- a/pkg/ddevapp/apptypes.go
+++ b/pkg/ddevapp/apptypes.go
@@ -98,6 +98,7 @@ func init() {
 			configOverrideAction: django4ConfigOverrideAction,
 			postConfigAction:     django4PostConfigAction,
 			postStartAction:      django4PostStartAction,
+			importFilesAction:    genericImportFilesAction,
 		},
 
 		nodeps.AppTypeDrupal6: {
@@ -159,6 +160,7 @@ func init() {
 			appTypeDetect:        isLaravelApp,
 			postStartAction:      laravelPostStartAction,
 			configOverrideAction: laravelConfigOverrideAction,
+			importFilesAction:    genericImportFilesAction,
 		},
 
 		nodeps.AppTypeMagento: {
@@ -181,13 +183,14 @@ func init() {
 
 		nodeps.AppTypePHP: {
 			postStartAction:   phpPostStartAction,
-			importFilesAction: phpImportFilesAction,
+			importFilesAction: genericImportFilesAction,
 		},
 
 		nodeps.AppTypePython: {
 			appTypeDetect:        isPythonApp,
 			configOverrideAction: pythonConfigOverrideAction,
 			postConfigAction:     pythonPostConfigAction,
+			importFilesAction:    genericImportFilesAction,
 		},
 
 		nodeps.AppTypeShopware6: {

--- a/pkg/ddevapp/ddevapp.go
+++ b/pkg/ddevapp/ddevapp.go
@@ -863,7 +863,7 @@ func (app *DdevApp) ImportFiles(uploadDir, importPath, extractPath string) error
 	if uploadDir == "" {
 		uploadDir = app.GetUploadDir()
 		if uploadDir == "" {
-			return fmt.Errorf("upload_dir is not set, cannot import files")
+			return fmt.Errorf("upload_dirs is not set, cannot import files")
 		}
 	}
 

--- a/pkg/ddevapp/ddevapp.go
+++ b/pkg/ddevapp/ddevapp.go
@@ -2847,3 +2847,48 @@ func FormatSiteStatus(status string) string {
 	}
 	return formattedStatus
 }
+
+// genericImportFilesAction defines the workflow for importing project files.
+func genericImportFilesAction(app *DdevApp, uploadDir, importPath, extPath string) error {
+	destPath := app.calculateHostUploadDirFullPath(uploadDir)
+
+	// parent of destination dir should exist
+	if !fileutil.FileExists(filepath.Dir(destPath)) {
+		return fmt.Errorf("unable to import to %s: parent directory does not exist", destPath)
+	}
+
+	// parent of destination dir should be writable.
+	if err := os.Chmod(filepath.Dir(destPath), 0755); err != nil {
+		return err
+	}
+
+	// If the destination path exists, remove it as was warned
+	if fileutil.FileExists(destPath) {
+		if err := os.RemoveAll(destPath); err != nil {
+			return fmt.Errorf("failed to cleanup %s before import: %v", destPath, err)
+		}
+	}
+
+	if isTar(importPath) {
+		if err := archive.Untar(importPath, destPath, extPath); err != nil {
+			return fmt.Errorf("failed to extract provided archive: %v", err)
+		}
+
+		return nil
+	}
+
+	if isZip(importPath) {
+		if err := archive.Unzip(importPath, destPath, extPath); err != nil {
+			return fmt.Errorf("failed to extract provided archive: %v", err)
+		}
+
+		return nil
+	}
+
+	//nolint: revive
+	if err := fileutil.CopyDir(importPath, destPath); err != nil {
+		return err
+	}
+
+	return nil
+}

--- a/pkg/ddevapp/ddevapp.go
+++ b/pkg/ddevapp/ddevapp.go
@@ -862,6 +862,9 @@ func (app *DdevApp) ImportFiles(uploadDir, importPath, extractPath string) error
 
 	if uploadDir == "" {
 		uploadDir = app.GetUploadDir()
+		if uploadDir == "" {
+			return fmt.Errorf("upload_dir is not set, cannot import files")
+		}
 	}
 
 	if err := app.dispatchImportFilesAction(uploadDir, importPath, extractPath); err != nil {

--- a/pkg/ddevapp/ddevapp_test.go
+++ b/pkg/ddevapp/ddevapp_test.go
@@ -165,8 +165,8 @@ var (
 		// 8: drupal9
 		{
 			Name:                          "TestPkgDrupal9",
-			SourceURL:                     "https://ftp.drupal.org/files/projects/drupal-9.4.2.tar.gz",
-			ArchiveInternalExtractionPath: "drupal-9.4.2/",
+			SourceURL:                     "https://ftp.drupal.org/files/projects/drupal-9.5.10.tar.gz",
+			ArchiveInternalExtractionPath: "drupal-9.5.10/",
 			FilesTarballURL:               "https://github.com/ddev/ddev_test_tarballs/releases/download/v1.1/d9_umami_files.tgz",
 			FilesZipballURL:               "https://github.com/ddev/ddev_test_tarballs/releases/download/v1.1/d9_umami_files.zip",
 			DBTarURL:                      "https://github.com/ddev/ddev_test_tarballs/releases/download/v1.1/d9_umami_sql.tar.gz",
@@ -225,8 +225,8 @@ var (
 		// 12: drupal10
 		{
 			Name:                          "TestPkgDrupal10",
-			SourceURL:                     "https://ftp.drupal.org/files/projects/drupal-10.0.0-alpha6.tar.gz",
-			ArchiveInternalExtractionPath: "drupal-10.0.0-alpha6",
+			SourceURL:                     "https://ftp.drupal.org/files/projects/drupal-10.1.1.tar.gz",
+			ArchiveInternalExtractionPath: "drupal-10.1.1/",
 			FilesTarballURL:               "https://github.com/ddev/ddev_test_tarballs/releases/download/v1.1/drupal10-files.tgz",
 			DBTarURL:                      "https://github.com/ddev/ddev_test_tarballs/releases/download/v1.1/drupal10-alpha6.sql.tar.gz",
 			FullSiteTarballURL:            "",

--- a/pkg/ddevapp/ddevapp_test.go
+++ b/pkg/ddevapp/ddevapp_test.go
@@ -2434,8 +2434,6 @@ func TestDdevImportFilesCustomUploadDir(t *testing.T) {
 		err := app.Init(site.Dir)
 		require.NoError(t, err)
 
-		// Try custom upload dir
-		//app.UploadDirs = ddevapp.UploadDirs{"my/upload/dir"}
 		absUploadDir := filepath.Join(app.AppRoot, app.Docroot, app.GetUploadDir())
 		err = os.MkdirAll(absUploadDir, 0755)
 		assert.NoError(err)

--- a/pkg/ddevapp/ddevapp_test.go
+++ b/pkg/ddevapp/ddevapp_test.go
@@ -2428,20 +2428,29 @@ func TestDdevImportFilesCustomUploadDir(t *testing.T) {
 
 	for _, site := range TestSites {
 		switchDir := site.Chdir()
-		runTime := util.TimeTrackC(fmt.Sprintf("%s %s", site.Name, t.Name()))
 		t.Logf("== BEGIN TestDdevImportFilesCustomUploadDir for %s\n", site.Name)
 
 		testcommon.ClearDockerEnv()
 		err := app.Init(site.Dir)
-		assert.NoError(err)
+		require.NoError(t, err)
 
-		// Set custom upload dir
-		app.UploadDirs = ddevapp.UploadDirs{"my/upload/dir"}
+		// Try custom upload dir
+		//app.UploadDirs = ddevapp.UploadDirs{"my/upload/dir"}
 		absUploadDir := filepath.Join(app.AppRoot, app.Docroot, app.GetUploadDir())
 		err = os.MkdirAll(absUploadDir, 0755)
 		assert.NoError(err)
 
+		// Reset to use the default upload_dirs for the project
+		err = app.Init(site.Dir)
+		require.NoError(t, err)
+
 		if site.FilesTarballURL != "" {
+			// First, try import with the project-default upload_dirs
+			// Make sure we don't have files to start
+			err = os.RemoveAll(app.GetHostUploadDirFullPath())
+			require.NoError(t, err)
+
+			err = os.RemoveAll(app.GetUploadDir())
 			_, tarballPath, err := testcommon.GetCachedArchive(site.Name, "local-tarballs-files", "", site.FilesTarballURL)
 			require.NoError(t, err)
 			err = app.ImportFiles("", tarballPath, "")
@@ -2449,6 +2458,35 @@ func TestDdevImportFilesCustomUploadDir(t *testing.T) {
 
 			// Ensure upload dir isn't empty
 			dirEntrySlice, err := os.ReadDir(absUploadDir)
+			assert.NoError(err)
+			assert.NotEmpty(dirEntrySlice)
+
+			// Second, try with a single custom upload_dirs
+			app.UploadDirs = ddevapp.UploadDirs{"my/upload/dir"}
+			err = os.MkdirAll("my/upload/dir", 0755)
+			require.NoError(t, err)
+			err = os.RemoveAll(app.GetHostUploadDirFullPath())
+			require.NoError(t, err)
+			_, tarballPath, err = testcommon.GetCachedArchive(site.Name, "local-tarballs-files", "", site.FilesTarballURL)
+			require.NoError(t, err)
+			err = app.ImportFiles("", tarballPath, "")
+			assert.NoError(err)
+			dirEntrySlice, err = os.ReadDir(app.GetHostUploadDirFullPath())
+			assert.NoError(err)
+			assert.NotEmpty(dirEntrySlice)
+
+			// Now try explicit import to targeted import_dir
+			app.UploadDirs = ddevapp.UploadDirs{"sites/default/files", "my/targeted/dir"}
+			targetDir := filepath.Join(app.AppRoot, "my/targeted/dir")
+			err = os.RemoveAll(targetDir)
+			require.NoError(t, err)
+			err = os.MkdirAll("my/targeted/dir", 0755)
+			require.NoError(t, err)
+			_, tarballPath, err = testcommon.GetCachedArchive(site.Name, "local-tarballs-files", "", site.FilesTarballURL)
+			require.NoError(t, err)
+			err = app.ImportFiles("my/targeted/dir", tarballPath, "")
+			assert.NoError(err)
+			dirEntrySlice, err = os.ReadDir(targetDir)
 			assert.NoError(err)
 			assert.NotEmpty(dirEntrySlice)
 		}
@@ -2477,7 +2515,12 @@ func TestDdevImportFilesCustomUploadDir(t *testing.T) {
 			assert.NotEmpty(dirEntrySlice)
 		}
 
-		runTime()
+		// We should not be able to import with upload_dirs=false
+		app.UploadDirs = false
+		err = app.ImportFiles("", "/tmp", "")
+		assert.Error(err)
+		require.Contains(t, err.Error(), "is not set for this project")
+
 		switchDir()
 	}
 }

--- a/pkg/ddevapp/ddevapp_test.go
+++ b/pkg/ddevapp/ddevapp_test.go
@@ -2133,7 +2133,7 @@ func TestDdevFullSiteSetup(t *testing.T) {
 			}
 			err = app.ImportFiles("", tarballPath, "")
 			assert.Error(err)
-			assert.Contains(err.Error(), fmt.Sprintf("upload_dirs is not set for this project (%s)", app.Type))
+			assert.Contains(err.Error(), "upload_dirs is not set", app.Type)
 		}
 		// We don't want all the projects running at once.
 		err = app.Stop(true, false)

--- a/pkg/ddevapp/ddevapp_test.go
+++ b/pkg/ddevapp/ddevapp_test.go
@@ -2447,8 +2447,8 @@ func TestDdevImportFilesCustomUploadDir(t *testing.T) {
 		if site.FilesTarballURL != "" {
 			// First, try import with the project-default upload_dirs
 			// Make sure we don't have files to start
-			targetFilesPath := app.GetUploadDir()
-			err = os.RemoveAll(targetFilesPath)
+			fullTargetFilesPath := app.GetHostUploadDirFullPath()
+			err = os.RemoveAll(fullTargetFilesPath)
 			require.NoError(t, err)
 
 			_, tarballPath, err := testcommon.GetCachedArchive(site.Name, "local-tarballs-files", "", site.FilesTarballURL)
@@ -2457,15 +2457,15 @@ func TestDdevImportFilesCustomUploadDir(t *testing.T) {
 			assert.NoError(err)
 
 			// Ensure upload dir isn't empty
-			dirEntrySlice, err := os.ReadDir(targetFilesPath)
+			dirEntrySlice, err := os.ReadDir(fullTargetFilesPath)
 			assert.NoError(err)
 			assert.NotEmpty(dirEntrySlice)
 
 			// Second, try with a single custom upload_dirs
-			targetFilesPath = "single/custom/target/dir"
+			targetFilesPath := "single/custom/target/dir"
 			// This is automatically relative to docroot
 			app.UploadDirs = ddevapp.UploadDirs{targetFilesPath}
-			fullTargetFilesPath := app.GetHostUploadDirFullPath()
+			fullTargetFilesPath = app.GetHostUploadDirFullPath()
 			err = os.MkdirAll(fullTargetFilesPath, 0755)
 			require.NoError(t, err)
 			err = fileutil.PurgeDirectory(fullTargetFilesPath)
@@ -2494,6 +2494,26 @@ func TestDdevImportFilesCustomUploadDir(t *testing.T) {
 			dirEntrySlice, err = os.ReadDir(secondTargetedFulPath)
 			assert.NoError(err)
 			assert.NotEmpty(dirEntrySlice)
+
+			// Try with a relative files dir, like ../private
+			if app.Docroot != "" {
+				targetFilesPath = "../private"
+				// This is automatically relative to docroot
+				app.UploadDirs = ddevapp.UploadDirs{targetFilesPath}
+				fullTargetFilesPath = app.GetHostUploadDirFullPath()
+				err = os.MkdirAll(fullTargetFilesPath, 0755)
+				require.NoError(t, err)
+				err = fileutil.PurgeDirectory(fullTargetFilesPath)
+				require.NoError(t, err)
+
+				_, tarballPath, err = testcommon.GetCachedArchive(site.Name, "local-tarballs-files", "", site.FilesTarballURL)
+				require.NoError(t, err)
+				err = app.ImportFiles("", tarballPath, "")
+				assert.NoError(err)
+				dirEntrySlice, err = os.ReadDir(fullTargetFilesPath)
+				assert.NoError(err)
+				assert.NotEmpty(dirEntrySlice)
+			}
 		}
 
 		if site.FilesZipballURL != "" {

--- a/pkg/ddevapp/ddevapp_test.go
+++ b/pkg/ddevapp/ddevapp_test.go
@@ -2447,46 +2447,51 @@ func TestDdevImportFilesCustomUploadDir(t *testing.T) {
 		if site.FilesTarballURL != "" {
 			// First, try import with the project-default upload_dirs
 			// Make sure we don't have files to start
-			err = os.RemoveAll(app.GetHostUploadDirFullPath())
+			targetFilesPath := app.GetUploadDir()
+			err = os.RemoveAll(targetFilesPath)
 			require.NoError(t, err)
 
-			err = os.RemoveAll(app.GetUploadDir())
 			_, tarballPath, err := testcommon.GetCachedArchive(site.Name, "local-tarballs-files", "", site.FilesTarballURL)
 			require.NoError(t, err)
 			err = app.ImportFiles("", tarballPath, "")
 			assert.NoError(err)
 
 			// Ensure upload dir isn't empty
-			dirEntrySlice, err := os.ReadDir(absUploadDir)
+			dirEntrySlice, err := os.ReadDir(targetFilesPath)
 			assert.NoError(err)
 			assert.NotEmpty(dirEntrySlice)
 
 			// Second, try with a single custom upload_dirs
-			app.UploadDirs = ddevapp.UploadDirs{"my/upload/dir"}
-			err = os.MkdirAll("my/upload/dir", 0755)
+			targetFilesPath = "single/custom/target/dir"
+			// This is automatically relative to docroot
+			app.UploadDirs = ddevapp.UploadDirs{targetFilesPath}
+			fullTargetFilesPath := app.GetHostUploadDirFullPath()
+			err = os.MkdirAll(fullTargetFilesPath, 0755)
 			require.NoError(t, err)
-			err = os.RemoveAll(app.GetHostUploadDirFullPath())
+			err = fileutil.PurgeDirectory(fullTargetFilesPath)
 			require.NoError(t, err)
+
 			_, tarballPath, err = testcommon.GetCachedArchive(site.Name, "local-tarballs-files", "", site.FilesTarballURL)
 			require.NoError(t, err)
 			err = app.ImportFiles("", tarballPath, "")
 			assert.NoError(err)
-			dirEntrySlice, err = os.ReadDir(app.GetHostUploadDirFullPath())
+			dirEntrySlice, err = os.ReadDir(fullTargetFilesPath)
 			assert.NoError(err)
 			assert.NotEmpty(dirEntrySlice)
 
 			// Now try explicit import to targeted import_dir
-			app.UploadDirs = ddevapp.UploadDirs{"sites/default/files", "my/targeted/dir"}
-			targetDir := filepath.Join(app.AppRoot, "my/targeted/dir")
-			err = os.RemoveAll(targetDir)
+			secondTargetDir := "second/targeted/dir"
+			secondTargetedFulPath := filepath.Join(app.AppRoot, app.Docroot, secondTargetDir)
+			app.UploadDirs = ddevapp.UploadDirs{"sites/default/files", secondTargetDir}
+			err = os.MkdirAll(secondTargetedFulPath, 0755)
 			require.NoError(t, err)
-			err = os.MkdirAll("my/targeted/dir", 0755)
+			err = fileutil.PurgeDirectory(secondTargetedFulPath)
 			require.NoError(t, err)
 			_, tarballPath, err = testcommon.GetCachedArchive(site.Name, "local-tarballs-files", "", site.FilesTarballURL)
 			require.NoError(t, err)
-			err = app.ImportFiles("my/targeted/dir", tarballPath, "")
+			err = app.ImportFiles(secondTargetDir, tarballPath, "")
 			assert.NoError(err)
-			dirEntrySlice, err = os.ReadDir(targetDir)
+			dirEntrySlice, err = os.ReadDir(secondTargetedFulPath)
 			assert.NoError(err)
 			assert.NotEmpty(dirEntrySlice)
 		}

--- a/pkg/ddevapp/ddevapp_test.go
+++ b/pkg/ddevapp/ddevapp_test.go
@@ -2444,7 +2444,7 @@ func TestDdevImportFilesCustomUploadDir(t *testing.T) {
 		err = app.Init(site.Dir)
 		require.NoError(t, err)
 
-		if site.FilesTarballURL != "" {
+		if site.FilesTarballURL != "" && site.UploadDirs != nil {
 			// First, try import with the project-default upload_dirs
 			// Make sure we don't have files to start
 			fullTargetFilesPath := app.GetHostUploadDirFullPath()

--- a/pkg/ddevapp/ddevapp_test.go
+++ b/pkg/ddevapp/ddevapp_test.go
@@ -2513,6 +2513,13 @@ func TestDdevImportFilesCustomUploadDir(t *testing.T) {
 				dirEntrySlice, err = os.ReadDir(fullTargetFilesPath)
 				assert.NoError(err)
 				assert.NotEmpty(dirEntrySlice)
+
+				// Try with upload_dir that is outside project
+				app.UploadDirs = "../../nowhere"
+				_, tarballPath, err := testcommon.GetCachedArchive(site.Name, "local-tarballs-files", "", site.FilesTarballURL)
+				require.NoError(t, err)
+				err = app.ImportFiles("", tarballPath, "")
+				assert.Error(err)
 			}
 		}
 
@@ -2544,7 +2551,7 @@ func TestDdevImportFilesCustomUploadDir(t *testing.T) {
 		app.UploadDirs = false
 		err = app.ImportFiles("", "/tmp", "")
 		assert.Error(err)
-		require.Contains(t, err.Error(), "is not set for this project")
+		require.Contains(t, err.Error(), "cannot import files")
 
 		switchDir()
 	}

--- a/pkg/ddevapp/ddevapp_test.go
+++ b/pkg/ddevapp/ddevapp_test.go
@@ -3235,7 +3235,7 @@ func TestHttpsRedirection(t *testing.T) {
 	types := ddevapp.GetValidAppTypes()
 	webserverTypes := []string{nodeps.WebserverNginxFPM, nodeps.WebserverApacheFPM}
 	if os.Getenv("GOTEST_SHORT") != "" {
-		types = []string{nodeps.AppTypePHP, nodeps.AppTypeDrupal8}
+		types = []string{nodeps.AppTypePHP, nodeps.AppTypeDrupal10}
 		webserverTypes = []string{nodeps.WebserverNginxFPM, nodeps.WebserverApacheFPM}
 	}
 	for _, projectType := range types {

--- a/pkg/ddevapp/php.go
+++ b/pkg/ddevapp/php.go
@@ -2,11 +2,6 @@ package ddevapp
 
 import (
 	"fmt"
-	"os"
-	"path/filepath"
-
-	"github.com/ddev/ddev/pkg/archive"
-	"github.com/ddev/ddev/pkg/fileutil"
 )
 
 func phpPostStartAction(app *DdevApp) error {
@@ -15,50 +10,5 @@ func phpPostStartAction(app *DdevApp) error {
 			return fmt.Errorf("failed to write settings file %s: %v", app.SiteDdevSettingsFile, err)
 		}
 	}
-	return nil
-}
-
-// phpImportFilesAction defines the workflow for importing project files.
-func phpImportFilesAction(app *DdevApp, uploadDir, importPath, extPath string) error {
-	destPath := app.calculateHostUploadDirFullPath(uploadDir)
-
-	// parent of destination dir should exist
-	if !fileutil.FileExists(filepath.Dir(destPath)) {
-		return fmt.Errorf("unable to import to %s: parent directory does not exist", destPath)
-	}
-
-	// parent of destination dir should be writable.
-	if err := os.Chmod(filepath.Dir(destPath), 0755); err != nil {
-		return err
-	}
-
-	// If the destination path exists, remove it as was warned
-	if fileutil.FileExists(destPath) {
-		if err := os.RemoveAll(destPath); err != nil {
-			return fmt.Errorf("failed to cleanup %s before import: %v", destPath, err)
-		}
-	}
-
-	if isTar(importPath) {
-		if err := archive.Untar(importPath, destPath, extPath); err != nil {
-			return fmt.Errorf("failed to extract provided archive: %v", err)
-		}
-
-		return nil
-	}
-
-	if isZip(importPath) {
-		if err := archive.Unzip(importPath, destPath, extPath); err != nil {
-			return fmt.Errorf("failed to extract provided archive: %v", err)
-		}
-
-		return nil
-	}
-
-	//nolint: revive
-	if err := fileutil.CopyDir(importPath, destPath); err != nil {
-		return err
-	}
-
 	return nil
 }

--- a/pkg/ddevapp/upload_dirs.go
+++ b/pkg/ddevapp/upload_dirs.go
@@ -49,8 +49,8 @@ func (app *DdevApp) GetUploadDir() string {
 func (app *DdevApp) GetUploadDirs() UploadDirs {
 	err := app.validateUploadDirs()
 	if err != nil {
-		// Should never happen
-		panic(err)
+		util.Warning("Ignoring invalid upload_dirs value: %v", err)
+		return UploadDirs{}
 	}
 
 	if app.UploadDirDeprecated != "" {
@@ -63,8 +63,8 @@ func (app *DdevApp) GetUploadDirs() UploadDirs {
 		return UploadDirs{}
 	}
 
-	if len(app.UploadDirs.(UploadDirs)) > 0 {
-		return app.UploadDirs.(UploadDirs)
+	if dirs, ok := app.UploadDirs.(UploadDirs); ok && len(dirs) > 0 {
+		return dirs
 	}
 
 	appFuncs, ok := appTypeMatrix[app.GetType()]
@@ -198,10 +198,13 @@ func (app *DdevApp) createUploadDirsIfNecessary() {
 // validateUploadDirs validates and converts UploadDirs to app.UploadDirs
 // interface or if disabled to bool false and returns nil if succeeded or an
 // error if not.
+// app.UploadDirs must be one of:
+// - slice of string (possibly empty)
+// - boolean false
 func (app *DdevApp) validateUploadDirs() error {
-	if _, ok := app.UploadDirs.(UploadDirs); ok {
-		// Conversion was already done, nothing to do.
-		return nil
+	if raw, ok := app.UploadDirs.(UploadDirs); ok {
+		// User provided a list of strings, convert it to UploadDirs.
+		app.UploadDirs = raw
 	}
 
 	typeOfUploadDirs := reflect.TypeOf(app.UploadDirs)
@@ -215,13 +218,7 @@ func (app *DdevApp) validateUploadDirs() error {
 		// User provided a string, convert it to UploadDirs.
 		app.UploadDirs = UploadDirs{app.UploadDirs.(string)}
 	case typeOfUploadDirs.Kind() == reflect.Slice:
-		// User provided a list of strings, convert it to UploadDirs.
-		uploadDirsRaw := app.UploadDirs.([]any)
-		uploadDirs := make(UploadDirs, 0, len(uploadDirsRaw))
-		for _, v := range uploadDirsRaw {
-			uploadDirs = append(uploadDirs, v.(string))
-		}
-		app.UploadDirs = uploadDirs
+		break
 	default:
 		// Provided value is not valid, user has to fix it.
 		return fmt.Errorf("`upload_dirs` must be a string, a list of strings, or false but `%v` given", app.UploadDirs)

--- a/pkg/ddevapp/upload_dirs.go
+++ b/pkg/ddevapp/upload_dirs.go
@@ -46,6 +46,8 @@ func (app *DdevApp) GetUploadDir() string {
 }
 
 // GetUploadDirs returns the upload (public files) directories.
+// These are gathered from the per-CMS configurations and the
+// value of upload_dirs. upload_dirs overrides the per-CMS configuration
 func (app *DdevApp) GetUploadDirs() UploadDirs {
 	err := app.validateUploadDirs()
 	if err != nil {
@@ -59,12 +61,31 @@ func (app *DdevApp) GetUploadDirs() UploadDirs {
 		app.addUploadDir(uploadDirDeprecated)
 	}
 
-	if app.UploadDirs == false {
+	switch app.UploadDirs.(type) {
+	case UploadDirs:
+		if len(app.UploadDirs.(UploadDirs)) > 0 {
+			return app.UploadDirs.(UploadDirs)
+		}
+		// Otherwise we go get the CMS-defined values
+	case []any:
+		if len(app.UploadDirs.([]any)) > 0 {
+			// User provided a list of strings, convert it to UploadDirs.
+			uploadDirsRaw := app.UploadDirs.([]any)
+			uploadDirs := make(UploadDirs, 0, len(uploadDirsRaw))
+			for _, v := range uploadDirsRaw {
+				uploadDirs = append(uploadDirs, v.(string))
+			}
+			app.UploadDirs = uploadDirs
+			return uploadDirs
+		}
+		// Otherwise we go get the CMS-defined values
+	case bool:
+		if app.UploadDirs.(bool) == false {
+			return UploadDirs{}
+		}
+	default:
+		util.Warning("app.UploadDirs is of invalid type %T", app.UploadDirs)
 		return UploadDirs{}
-	}
-
-	if dirs, ok := app.UploadDirs.(UploadDirs); ok && len(dirs) > 0 {
-		return dirs
 	}
 
 	appFuncs, ok := appTypeMatrix[app.GetType()]

--- a/pkg/ddevapp/upload_dirs.go
+++ b/pkg/ddevapp/upload_dirs.go
@@ -35,6 +35,7 @@ func (app *DdevApp) addUploadDir(uploadDir string) {
 }
 
 // GetUploadDir returns the first upload (public files) directory.
+// This value is relative to the docroot
 func (app *DdevApp) GetUploadDir() string {
 	uploadDirs := app.GetUploadDirs()
 	if len(uploadDirs) > 0 {

--- a/pkg/ddevapp/upload_dirs.go
+++ b/pkg/ddevapp/upload_dirs.go
@@ -194,7 +194,7 @@ func (app *DdevApp) createUploadDirsIfNecessary() {
 	}
 }
 
-// validateUploadDirs validates and converts UploadDirs to a app.UploadDirs
+// validateUploadDirs validates and converts UploadDirs to app.UploadDirs
 // interface or if disabled to bool false and returns nil if succeeded or an
 // error if not.
 func (app *DdevApp) validateUploadDirs() error {
@@ -226,10 +226,12 @@ func (app *DdevApp) validateUploadDirs() error {
 		return fmt.Errorf("`upload_dirs` must be a string, a list of strings, or false but `%v` given", app.UploadDirs)
 	}
 
-	// Check upload dirs are in the project root.
-	for _, uploadDir := range app.UploadDirs.(UploadDirs) {
-		if !strings.HasPrefix(app.calculateHostUploadDirFullPath(uploadDir), app.AppRoot) {
-			return fmt.Errorf("invalid upload dir `%s` outside of project root `%s` found", uploadDir, app.AppRoot)
+	if dirs, ok := app.UploadDirs.(UploadDirs); ok {
+		// Check upload dirs are in the project root.
+		for _, uploadDir := range dirs {
+			if !strings.HasPrefix(app.calculateHostUploadDirFullPath(uploadDir), app.AppRoot) {
+				return fmt.Errorf("invalid upload dir `%s` outside of project root `%s` found", uploadDir, app.AppRoot)
+			}
 		}
 	}
 

